streams/src/main/java/org/apache/kafka/streams/KafkaClientSupplier.java

```diff
@@ -35,8 +36,25 @@
      *
      * @param config Supplied by the {@link java.util.Properties} given to the {@link KafkaStreams}
      * @return an instance of {@link AdminClient}
+     * @deprecated Not called by Kafka Streams, which now uses {@link #getAdmin} instead.
      */
-    AdminClient getAdminClient(final Map<String, Object> config);
+    @Deprecated
+    default AdminClient getAdminClient(final Map<String, Object> config) {
+        throw new UnsupportedOperationException("Direct use of this method is deprecated. " +
+            "Implementations of KafkaClientSupplier should implement the getAdmin method instead. " +
```

Comment:
nit: `getAdmin method` -> `getAdmin() method` ?

Reply:
This is an incompatible change, why are we doing it in a minor release?

Reply:
I see, the assumption is that existing implementations will have overriden it.

Reply:
Yeah, it's not incompatible with our in-built impls or any existing user impls.